fl4health/utils/partitioners.py

```diff
@@ -70,17 +85,29 @@ def partition_label_indices(self, label: T, label_indices: torch.Tensor) -> Tupl
 
         Returns:
             List[torch.Tensor]: partitioned indices of datapoints with the corresponding label.
+            int: The minimum number of data points assigned to a partition.
+            np.ndarray: The Dirichlet distribution used to partition the data points.
         """
-        total_data_points_with_label = len(label_indices)
-        # These are the percentages of the label indices to be distributed for each partition
-        partition_allocations = np.random.dirichlet(np.repeat(self.beta, self.number_of_partitions))
-        log(
-            INFO,
-            (
-                f"The allocation distribution for label ({str(label)}) is {partition_allocations} "
-                f"using a beta of {self.beta}",
-            ),
-        )
+        if self.prior_distribution is not None:
+            assert (
+                len(self.prior_distribution) == self.n_unique_labels
+            ), "The length of the prior must match the number of labels"
+            partition_allocations = self.prior_distribution[label]
+            log(
+                INFO,
+                f"Using a prior distribution of {self.prior_distribution[label]} to allocate the label ({str(label)})",
```

Comment:
Super minor, but we already have `partition_allocations`, so we might as well use it for the logging here?

Reply:
Correct, I also updated wording to match the below logging.